graviti/operation/frame.py

```diff
@@ -113,6 +118,57 @@ def do(  # pylint: disable=invalid-name
             )
 
 
+class UpdateSchema(DataFrameOperation):
+    """This class defines the operation that update the schema of a DataFrame.
+
+    Arguments:
+        schema: New portex schema after updated.
+
+    """
+
+    def __init__(self, schema: PortexType) -> None:
+        self.schema = schema
+
+    def do(  # pylint: disable=invalid-name, unused-argument
+        self,
+        access_key: str,
+        url: str,
+        owner: str,
+        dataset: str,
+        *,
+        draft_number: int,
+        sheet: str,
+        jobs: int,
+    ) -> None:
+        """Execute the OpenAPI update schema.
+
+        Arguments:
+            access_key: User's access key.
+            url: The URL of the graviti website.
+            owner: The owner of the dataset.
+            dataset: Name of the dataset, unique for a user.
+            draft_number: The draft number.
+            sheet: The sheet name.
+            jobs: The number of the max workers in multi-thread operation.
+
+        """
+        schema = self.schema.to_yaml()
+        pyarrow_schema = pa.schema(self.schema.fields.to_pyarrow())  # type: ignore[attr-defined]
+        arrow_schema = base64.encodebytes(pyarrow_schema.serialize().to_pybytes()).decode("ascii")
+        avro_schema = json.dumps(convert_portex_schema_to_avro(self.schema))
```

Comment:
Does this part of logic exist in other operations as well? Can we reuse them?

Reply:
yes, CreateSheet operation also use this part. should we add a method to schema or add a function to sheet operation or frame operation